mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/MessageMapperTest.java

```diff
@@ -377,6 +378,23 @@ void deleteShouldNotUpdateRecentWhenNotNeeded() throws MailboxException {
         assertThat(messageMapper.findRecentMessageUidsInMailbox(benwaInboxMailbox)).containsOnly(message2.getUid(), message4.getUid());
     }
 
+    @Test
+    void deleteMessagesShouldCorrectlyUpdateMailboxCounters() throws MailboxException {
+        saveMessages();
+        messageMapper.updateFlags(benwaInboxMailbox, new FlagsUpdateCalculator(new Flags(Flag.SEEN), FlagsUpdateMode.REPLACE), MessageRange.one(message2.getUid()));
+        messageMapper.updateFlags(benwaInboxMailbox, new FlagsUpdateCalculator(new Flags(Flag.SEEN), FlagsUpdateMode.REPLACE), MessageRange.one(message3.getUid()));
+        messageMapper.updateFlags(benwaInboxMailbox, new FlagsUpdateCalculator(new Flags(Flag.SEEN), FlagsUpdateMode.REPLACE), MessageRange.one(message4.getUid()));
+
+        messageMapper.deleteMessages(benwaInboxMailbox, ImmutableList.of(message1.getUid(), message2.getUid(), message3.getUid()));
+
+        assertThat(messageMapper.getMailboxCounters(benwaInboxMailbox))
+            .isEqualTo(MailboxCounters.builder()
+                .mailboxId(benwaInboxMailbox.getMailboxId())
+                .count(2)
+                .unseen(1)
```

Comment:
sorry, this class is hard to read I don't get why you expect 1 unseen at the end

Reply:
Sure. 

We have 5 messages:
 - **message1**, **message2**, **message3**, are deleted
 -  **message4** is seen
 -  **message4** is unseen (created in saveMessages)

Hence the expected count. Note that this test runs on the memory version not relying on denormalisation.

Reply:
we should rename the test.
Let me try: "deleteMessagesShouldDecrementUnseenToOneWhenDeletingOneUnseenMessageOnTwo"
I think the test would be easier to understand without message 1, 2 and 3.